sqlmesh/core/engine_adapter/base.py

```diff
@@ -2483,6 +2515,35 @@ def _truncate_table(self, table_name: TableName) -> None:
         table = exp.to_table(table_name)
         self.execute(f"TRUNCATE TABLE {table.sql(dialect=self.dialect, identify=True)}")
 
+    def _drop_data_object_on_type_mismatch(
+        self, target_name: TableName, expected_type: DataObjectType
+    ) -> bool:
+        """Drops a data object if it exists and is not of the expected type.
+
+        Args:
+            target_name: The name of the data object to check.
+            expected_type: The expected type of the data object.
+
+        Returns:
+            True if the data object exists and is of the expected type, False otherwise.
```

Comment:
My instinct is that this should return a bool indicating if the drop was performed or not.

Reply:
I'm not sure what you mean by permitted. 

To help explain my point, the method is called `drop_data_object_on_type_mismatch` and returns a bool. I read the method as conditional and saying it is going to perform some action (drop) on some condition (type mismatch). Therefore I would expect the bool to indicate if the condition was true and the action was performed. Right now though it returns true if the condition is not true and the action was not performed which doesn't match my intuition.